### PR TITLE
Fix develop install

### PR DESCRIPTION
**Pull request by ultmaster**

### CODE REVIEW

File: setup_ts.py [hunk 1]
```diff
@@ -132,6 +133,7 @@ def prepare_nni_node():
     node_src = Path('toolchain/node', node_executable_in_tarball)
     node_dst = Path('nni_node', node_executable)
     shutil.copyfile(node_src, node_dst)
+    os.chmod(node_dst, os.stat(node_dst).st_mode | stat.S_IEXEC)
```
Comment (liuzhe-lz) [inline setup_ts.py:136]: `shutil.copymode()` should be better.

**Comment by liuzhe-lz** [comment]
Empty `__init__.py` should be unnecessary since Python 3.3.

File: setup.py [hunk 2]
```diff
@@ -164,7 +164,7 @@ def run(self):
 
 class Develop(develop):
     def finalize_options(self):
-        self.user = True  # always use `develop --user`
+        self.user = False  # always use `develop --user`
```
Comment (liuzhe-lz) [inline setup.py:167]: I prefer to explicitly check conda/venv/virtualenv.
Pip will not fallback to `--user` when python directory is protected, and the error message will not suggest to do so either.

If you don't want to do the check anyway, just remove this function. Forcing to `False` will make it impossible to install as normal user without venv.

**Comment by ultmaster** [comment]
> Empty `__init__.py` should be unnecessary since Python 3.3. (github cannot comment on empty file)

@liuzhe-lz I added them because on my machine, nnictl cannot be installed at all because `tools` does not have an `__init__.py` and all the files within are ignored. I'm using Python 3.7.5 with pip 19.3.1 under conda. Not sure why.

**Comment by ultmaster** [comment]
@liuzhe-lz will open a separate PR to resolve comments (and some other issues, possibly). 👍

**Comment by liuzhe-lz** [comment]
> > Empty `__init__.py` should be unnecessary since Python 3.3. (github cannot comment on empty file)
> 
> @liuzhe-lz I added them because on my machine, nnictl cannot be installed at all because `tools` does not have an `__init__.py` and all the files within are ignored. I'm using Python 3.7.5 with pip 19.3.1 under conda. Not sure why.

How did you install it? As mentioned in the [PR](https://github.com/microsoft/nni/pull/2987) you should use `python setup.py develop`, which will not involve pip.

**Comment by ultmaster** [comment]
The issue has been fixed in #3063.

**Pull request closed without merging**